src/Course/List.hs

```diff
@@ -390,31 +528,39 @@ seqOptional =
  > find (const True) infinity
  Full 0
 -}
+-- TODO: Doesn't work in the infinite case, can we use fold?
+-- find :: (a -> Bool) -> List a -> Optional a
+-- find f xs = foldLeft step init list
+--   where
+--     step = \acc next -> acc <+> if (f next) then Full next else Empty
+--     init = Empty
+--     list = xs
```

Comment:
Are you sure yours doesn't work in the infinite case? It looks to me like it does.

Reply:
The commented version blows up my stack, and then my computer. That was fun 😅 .